File: bindings/go/blazen.go
package blazen

import (
	uniffiblazen "github.com/zorpxinc/blazen-go/internal/uniffi/blazen"
)

// Init initializes the Blazen runtime.
//
// The first call sets up the embedded Tokio runtime that drives every
// async operation across the FFI boundary, configures default
// tracing-subscriber output (controlled by RUST_LOG), and prepares the
// native side for use. Subsequent calls are no-ops, so it is safe to
// call from multiple goroutines or initialisation paths.
//
// Most package functions call Init internally, so callers usually do not
// need to invoke it explicitly — but doing so eagerly avoids the small
// startup cost on the first real request.
func Init() {
	uniffiblazen.Init()
}

// Version returns the version of the embedded blazen-uniffi native
// library. Useful for diagnosing version-skew between the Go module and
// the bundled native lib.
func Version() string {
	return uniffiblazen.Version()
}

// Shutdown flushes telemetry exporters and any other shutdown work that
// the native runtime needs before the host process exits. This is
// idempotent and safe to call from a defer in a long-running program.
//
// If no telemetry exporters were initialised (InitLangfuse / InitOtlp /
// InitPrometheus), Shutdown is a no-op.
func Shutdown() {
	uniffiblazen.ShutdownTelemetry()
}

File: bindings/swift/Sources/BlazenSwift/Blazen.swift
import Foundation
import UniFFIBlazen

/// Top-level entry point for the Blazen Swift binding.
///
/// `Blazen` is a namespace of static helpers covering one-time process
/// lifecycle (`initialize`, `shutdown`) and version introspection.
/// Everything else lives on dedicated types: `Workflow`, `Pipeline`,
/// `CompletionModel`, `EmbeddingModel`, `Agent`, the `Providers` factories,
/// etc.
public enum Blazen {
    /// Run Blazen's one-time process initialisation.
    ///
    /// Safe to call multiple times — subsequent calls are no-ops. Brings up
    /// the shared Tokio runtime that backs every `async` method in this
    /// binding. Recommended at app launch (e.g. in `@main`) so the first
    /// request a user issues doesn't pay the runtime spin-up cost.
    public static func initialize() {
        UniFFIBlazen.`init`()
    }

    /// Best-effort flush + shutdown of any initialised telemetry exporters.
    ///
    /// Always safe to call, even when no exporters were initialised. Call
    /// once near process exit (e.g. inside an `atexit`-style hook) so any
    /// buffered telemetry has a chance to ship before the runtime is torn
    /// down.
    public static func shutdown() {
        UniFFIBlazen.shutdownTelemetry()
    }

    /// The semantic version of the underlying `blazen-uniffi` native lib,
    /// as compiled into the loaded `libblazen_uniffi` artefact.
    ///
    /// Useful for diagnosing version skew between the Swift wrapper and
    /// the native lib shipped alongside it.
    public static var version: String {
        UniFFIBlazen.version()
    }
}
